import sys
import datetime
import sqlite3
from sqlite3 import Error

precioF=0
respuestaW=1
respuesta=1
clave_venta=0

while respuesta==1:
    print("*** MENU ***")
    print("")
    print("[1] Registrar ventas")
    print("[2] Consultar ventas")
    print("[3] Reporte de fechas")
    print("[4] Salir")
    print("")
    seleccion=int(input("Ingrese una accion: \n"))
    
    if seleccion==1:
            try:
                with sqlite3.connect("Cosmeticos.db") as conn:
                    while respuestaW==1:
                        print("Conexi贸n establecida")
                        mi_cursor = conn.cursor()
                        clave_venta=input("Que folio de venta fue: \n")
                        fechaActual = datetime.datetime.combine(datetime.date.today(), datetime.datetime.min.time())
                        Articulo=input("Que articulo se ha vendido?: \n")
                        p=int(input("Cuantas piezas fueron vendidas?: \n"))
                        precio=int(input(f"A que precio fue vendido $ {Articulo}?: \n"))
                        precioFinal=(precio*p)
                        print(f"La cantidad a pagar es: {precioFinal}")
                        valores = {"Folio": clave_venta, "Fecha": fechaActual, "Articulo":Articulo, "Cantidad":p, "Precio":precio}
                        mi_cursor.execute("INSERT INTO RegistroVentas VALUES(:Folio, :Fecha, :Articulo, :Cantidad, :Precio)",valores)
                        print("Registro agregado exitosamente")
                        respuestaW=int(input("Desea registrar otra venta? [1] SI [0] NO?: "))
            except Error as e:
                print (e)
            except:
                print(f"Se produjo el siguiente error: {sys.exc_info()[0]}")
            finally:
                if (conn):
                    conn.close()
                    print("Se ha cerrado la conexi贸n")
    if seleccion==2:
        try:
            with sqlite3.connect("Cosmeticos.db") as conn:
                mi_cursor = conn.cursor()
                claveConsulta=int(input("Dime que clave quieres consultar: \n"))
                criterios = {"Folio":claveConsulta}
                mi_cursor.execute("SELECT Folio,Fecha,Descripcion,Cantidad,Precio FROM RegistroVentas WHERE Folio = :Folio;", criterios)
                registros = mi_cursor.fetchall()
   
                for Folio, Fecha, Articulo, Cantidad, Precio in registros:
                    print(f"Folio = {Folio}")
                    print(f"Fecha de registro = {Fecha}")
                    print(f"Descripcion = {Articulo}")
                    print(f"Cantidad = {Cantidad}")
                    print(f"Precio = {Precio}")
                respuestaW=int(input("Desea consultar otro dato? [1] SI [0] NO "))
        except sqlite3.Error as e:
            print (e)
        except Exception:
            print(f"Se produjo el siguiente error: {sys.exc_info()[0]}")
        finally:
            if (conn):
                conn.close()
                print("Se ha cerrado la conexi贸n")
                
    if seleccion==3:
        fecha_consultar = input("Dime una fecha (dd/mm/aaaa): ")
        fecha_consultar = datetime.datetime.strptime(fecha_consultar, "%d/%m/%Y").date()
        try:
            with sqlite3.connect("Cosmeticos.db", detect_types = sqlite3.PARSE_DECLTYPES | sqlite3.PARSE_COLNAMES) as conn:
                mi_cursor = conn.cursor()
                criterios = {"fecha":fecha_consultar}
                mi_cursor.execute("SELECT Folio, Fecha, Descripcion, Cantidad, Precio FROM RegistroVentas WHERE DATE(Fecha) = :fecha;", criterios)
                registros = mi_cursor.fetchall()
   
                for Folio, Fecha, Articulo, Cantidad, Precio in registros:
                    print(f"Folio = {Folio}")
                    print(f"Fecha de registro = {Fecha}")
                    print(f"Descripcion = {Articulo}")
                    print(f"Cantidad = {Cantidad}")
                    print(f"Precio = {Precio}")
                    print("*"*5)
            respuestaW=int(input("Desea consultar otro dato? [1] SI [0] NO "))
        except sqlite3.Error as e:
            print (e)
        except Exception:
            print(f"Se produjo el siguiente error: {sys.exc_info()[0]}")
        finally:
            if (conn):
                conn.close()
                print("Se ha cerrado la conexi贸n")
                
                
    if seleccion==4:
        print("Ha cerrado el programa con exito")
        break
